lib/opentelemetry/trace/exporter/http_client.lua

```diff
@@ -28,29 +38,51 @@ function _M.new(address, timeout, headers)
         uri = uri,
         timeout = timeout,
         headers = headers,
+        httpc = httpc,
+        use_gzip = use_gzip,
     }
+
     return setmetatable(self, mt)
 end
 
 function _M.do_request(self, body)
-    local httpc = http.new()
-    httpc:set_timeout(self.timeout * 1000)
+    self.httpc = self.httpc or http.new()
+    self.httpc:set_timeout(self.timeout * 1000)
+
+    if self.use_gzip then
+        -- Compress (deflate) request body
+        -- the compression should be set to Best Compression and window size
+        -- should be set to 15+16, see reference below:
+        -- https://github.com/brimworks/lua-zlib/issues/4#issuecomment-26383801
+        local deflate_stream = zlib.deflate(zlib.BEST_COMPRESSION, 15+16)
```

Comment:
I was thinking about this a bit more ... I think we should figure out how to do the compression in the otlp exporter's `call_collector` method. This is because requests to the collector can retry multiple times, and we don't want to re-gzip the payload each time. That's what we do with protobuf encoding (see https://github.com/yangxikun/opentelemetry-lua/blob/82ff258dffc4711511f11790c4336a9315649fd5/lib/opentelemetry/trace/exporter/otlp.lua#L58 and https://github.com/yangxikun/opentelemetry-lua/blob/82ff258dffc4711511f11790c4336a9315649fd5/lib/opentelemetry/trace/exporter/otlp.lua#L104)

Reply:
sounds good, shouldn't be hard :+1:

Reply:
I moved the compression to the otlp exporter and adjusted the tests, I think it looks good, I am a bit worried that it pushes complexity [into the worker init block](https://github.com/yangxikun/opentelemetry-lua/pull/87/files#diff-3ea8ef5dc8ca1c4a7711c4a9b0de70c5280788adc043abfbc1b2afe6dc9b44ca) but I'll you be the judge of that.